ff/serving/metadata/dashboard/dashboard_metadata.go

```diff
@@ -155,7 +155,183 @@ func (m *FetchError) Error() string {
+			trainingSetVariantMap[variant.Variant()] = trainingSetShallowMap(variant)
+			trainingSetMap[variant.Name()] = TrainingSetResource{
+				AllVariants:    []string{variant.Variant()},
+				Type:           "TrainingSet",
+				DefaultVariant: "",
+				Name:           variant.Name(),
+				Variants:       trainingSetVariantMap,
+			}
+		}
+	}
+	var trainingSetList []TrainingSetResource
+	for _, value := range trainingSetMap {
+		trainingSetList = append(trainingSetList, value)
+	}
+	return trainingSetList, nil
+}
+
+func (m *MetadataServer) nameVariantToFeatureList(nameVariants []metadata.NameVariant) ([]FeatureResource, error) {
+	featureMap := make(map[string]FeatureResource)
+	featureVariants, err := m.client.GetFeatureVariants(context.Background(), nameVariants)
+	if err != nil {
+		return nil, err
+	}
+	for _, variant := range featureVariants {
+
+		if featureResource, has := featureMap[variant.Name()]; has {
+			featureResource.AllVariants = append(featureMap[variant.Name()].AllVariants, variant.Variant())
+			featureResource.Variants[variant.Variant()] = featureShallowMap(variant)
+			featureMap[variant.Name()] = featureResource
+		} else {
+			featureVariantMap := make(map[string]FeatureVariantResource)
+			featureVariantMap[variant.Variant()] = featureShallowMap(variant)
+			featureMap[variant.Name()] = FeatureResource{
+				AllVariants:    []string{variant.Variant()},
+				Type:           "Feature",
+				DefaultVariant: "",
+				Name:           variant.Name(),
+				Variants:       featureVariantMap,
+			}
+		}
+	}
+	var featureList []FeatureResource
+	for _, value := range featureMap {
+		featureList = append(featureList, value)
+	}
+	return featureList, nil
+}
+
+func (m *MetadataServer) nameVariantToLabelList(nameVariants []metadata.NameVariant) ([]LabelResource, error) {
+	labelMap := make(map[string]LabelResource)
+	labelVariants, err := m.client.GetLabelVariants(context.Background(), nameVariants)
+	if err != nil {
+		return nil, err
+	}
+	for _, variant := range labelVariants {
+
+		if labelResource, has := labelMap[variant.Name()]; has {
+			labelResource.AllVariants = append(labelMap[variant.Name()].AllVariants, variant.Variant())
+			labelResource.Variants[variant.Variant()] = labelShallowMap(variant)
+			labelMap[variant.Name()] = labelResource
+		} else {
+			labelVariantMap := make(map[string]LabelVariantResource)
+			labelVariantMap[variant.Variant()] = labelShallowMap(variant)
+			labelMap[variant.Name()] = LabelResource{
+				AllVariants:    []string{variant.Variant()},
+				Type:           "Label",
+				DefaultVariant: "",
+				Name:           variant.Name(),
+				Variants:       labelVariantMap,
+			}
+		}
+	}
+	var labelList []LabelResource
+	for _, value := range labelMap {
+		labelList = append(labelList, value)
+	}
+	return labelList, nil
+}
+
+func (m *MetadataServer) nameVariantToSourceList(nameVariants []metadata.NameVariant) ([]SourceResource, error) {
```

Comment:
Make this return a []SourceVariantResource or map[string][]SourceVariantResource